tests/Admin/BaseFieldDescriptionTest.php

```diff
@@ -162,101 +161,109 @@ public function testAdmin(): void
-            $mock3 = $this->getMockBuilder(\stdClass::class)
-                ->setMethods([$method])
-                ->getMock();
-
-            $mock3->expects($this->once())->method($method)->willReturn(42);
-            $this->assertSame(42, $description3->getFieldValue($mock3, '_fake'));
-        }
-
-        $mock4 = $this->getMockBuilder('MockedTestObject')
-            ->setMethods(['myMethod'])
-            ->getMock();
-        $mock4->expects($this->once())
-            ->method('myMethod')
-            ->willReturn('myMethodValue');
-
-        $description4 = new FieldDescription('name');
-        $description4->setOption('code', 'myMethod');
+        $description2 = new FieldDescription('name', [
+            'code' => 'getWithTwoParameters',
+            'parameters' => [$arg1, $arg2],
+        ]);
 
-        $this->assertSame($description4->getFieldValue($mock4, null), 'myMethodValue');
+        $mock2 = $this->getMockBuilder(\stdClass::class)->addMethods(['getWithTwoParameters'])->getMock();
+        $mock2->method('getWithTwoParameters')->with($arg1, $arg2)->willReturn($arg1 + $arg2);
+        $this->assertSame(42, $description2->getFieldValue($mock2, 'fake'));
     }
 
-    public function testGetValueNoValueException(): void
+    public function testGetFieldValueWithMagicCall(): void
     {
-        $this->expectException(\Sonata\AdminBundle\Exception\NoValueException::class);
+        $parameters = ['foo', 'bar'];
+        $foo = new FooCall();
 
         $description = new FieldDescription('name');
-        $mock = $this->getMockBuilder(\stdClass::class)
-            ->setMethods(['getFoo'])
-            ->getMock();
+        $description->setOption('parameters', $parameters);
+        $this->assertSame(['fake', $parameters], $description->getFieldValue($foo, 'fake'));
 
-        $description->getFieldValue($mock, 'fake');
+        // repeating to cover retrieving cached getter
+        $this->assertSame(['fake', $parameters], $description->getFieldValue($foo, 'fake'));
     }
 
     /**
-     * @doesNotPerformAssertions
+     * @dataProvider getFieldValueWithFieldNameDataProvider
      */
-    public function testGetVirtualValue(): void
+    public function testGetFieldValueWithMethod(string $method): void
     {
         $description = new FieldDescription('name');
-        $mock = $this->getMockBuilder(\stdClass::class)
-            ->setMethods(['getFoo'])
-            ->getMock();
+        $mock = $this->getMockBuilder(\stdClass::class)->addMethods([$method])->getMock();
 
-        $description->setOption('virtual_field', true);
-        $description->getFieldValue($mock, 'fake');
+        $mock->method($method)->willReturn(42);
+        $this->assertSame(42, $description->getFieldValue($mock, 'fake_field_value'));
+        $this->assertSame(42, $description->getFieldValue($mock, 'fakeFieldValue'));
+    }
+
+    public function getFieldValueWithFieldNameDataProvider(): iterable
```

Comment:
I don't remember which prefix we use in Sonata (if we still use one), but I think it's great to document the return value of data providers to match it against the signature of tests that use them.

```suggestion
    /**
     * @return iterable<array{string}>
     */
    public function getFieldValueWithFieldNameDataProvider(): iterable
```

Reply:
Done